Mathlib/FieldTheory/Galois/InfiniteDimensional.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2024 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan, Yuyang Zhao, Jujian Zhang
+-/
+
+import Mathlib.Algebra.Category.Grp.FiniteGrp
+import Mathlib.CategoryTheory.Category.Preorder
+import Mathlib.FieldTheory.NormalClosure
+import Mathlib.FieldTheory.SeparableClosure
+
+/-!
+
+# Main definitions and results
+
+In a field extension `K/k`
+
+* `FiniteGaloisIntermediateField` : The type of a finite Galois intermediate field of `K/k`
+
+* `adjoin` : The finite Galois intermediate field obtained from the normal closure of adjoining a
+  `s : Set K` to `k`.
+
+* `finGaloisGroup L` : The (finite) Galois group `Gal(K / k)` associated to a
+  `L : FiniteGaloisIntermediateField k K` `L`.
+
+* `finGaloisGroupMap` : For `FiniteGaloisIntermediateField` s `L₁` and `L₂` with `L₂ ≤ L₁`
+  giving the restriction of `Gal(L₁/k)` to `Gal(L₂/k)`
+
+* `finGaloisGroupFunctor` : Mapping `FiniteGaloisIntermediateField` ordered by inverse inclusion to
+  its corresponding Galois Group as `FiniteGrp`.
+
+# TODO
+
+* `FiniteGaloisIntermediateField` should be a `ConditionallyCompleteLattice` but isn't proved yet.
+
+-/
+
+open CategoryTheory Opposite
+open scoped IntermediateField
+
+variable (k K : Type*) [Field k] [Field K] [Algebra k K]
+
+/-- The type of a finite Galois intermediate field of `K/k` -/
+@[ext]
```

Comment:
Is this `ext` attribute useful? Both fields are `Prop`s, so I think it is useless. I tried it on an example and I do not think it has any effect. Have you good reasons for adding it?

Reply:
No, I would try to remove it.